Made the list of orgs UI

divy-arpit opened the pull request:
### Fixes #27 

### Description

Basic UI for the app to show the list of organisations of which user is a part.

**Type of Change:** (Delete irrelevant options)
- Code
- User Interface


**Code/Quality Assurance Only** (Delete irrelevant options)



- New feature (non-breaking change which adds functionality pre-approved by mentors)


### How Has This Been Tested?

Tested on local machine



### Checklist:

- [X] My PR follows the style guidelines of this project
- [X] I have performed a self-review of my own code or materials
- [X] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] Any dependent changes have been merged


**Code/Quality Assurance**
- [ ] My code follows TDD clean architecture
- [X] My changes generate no new warnings 
- [ ] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
![Screenshot_1590721960](https://user-images.githubusercontent.com/61823356/83217613-cbcca780-a189-11ea-9255-371966cbc436.png)

divy-arpit committed:
Made the list of orgs UI

werainkhatri commented on lib/screens/listOfOrgs/listOfOrgsScreen.dart:
```diff
@@ -0,0 +1,24 @@
+import 'package:flutter/material.dart';
+import 'package:teesco/screens/listOfOrgs/widgets.dart';
+
+class ListOfOrganisations extends StatefulWidget {
```
1. Try to name the class compactly
```suggestion
class OrganisationList extends StatefulWidget {
```
2. Make sure the name of a widget ends with what it actually is
```suggestion
class OrganizationListScreen extends StatefulWidget {
```
No need to add screen in the name of the file, but add it in the name of the class

werainkhatri commented on lib/screens/listOfOrgs/listOfOrgsScreen.dart:
```diff
@@ -0,0 +1,24 @@
+import 'package:flutter/material.dart';
```
Please follow the naming conversion...
1. file and folder names: `lower_case_words_separated_by_underscores`
2. class names: `UpperCamelCase`
3. variables and functions: `lowerCamelCase`

werainkhatri commented on lib/screens/listOfOrgs/widgets.dart:
```diff
@@ -0,0 +1,93 @@
+import 'package:flutter/material.dart';
```
widgets should be a folder containing files of stateless/stateful widgets
not a single file containing all widgets

werainkhatri commented on lib/screens/listOfOrgs/widgets.dart:
```diff
@@ -0,0 +1,93 @@
+import 'package:flutter/material.dart';
+import 'package:flutter/rendering.dart';
+
+Widget appBar() {
```
these need to be stateless/stateful widget classes (decided by you)
in their separate files

werainkhatri commented on lib/screens/listOfOrgs/listOfOrgsScreen.dart:
```diff
@@ -0,0 +1,24 @@
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: appBar(),
```
you need to create a separate widget only if the widget is very big
also, go through the appbar [documentation](https://api.flutter.dev/flutter/material/AppBar-class.html) and [examples](https://flutter.dev/docs/catalog/samples/basic-app-bar) for better understanding of how appbar should be used

werainkhatri reviewed (changes requested):
1. Write proper commits messages... refer [commit guide](https://https://github.com/ECellNitrr/teesco-app/wiki/Commit-Message-Structure) for instructions
2. Follow naming convention. (first review on listOfOrgs.dart file)
3. Read documentation for various flutter widgets with their usecase. go through the [widget of the week](https://www.youtube.com/watch?v=b_sQ9bMltGU&list=PLjxrf2q8roU23XGwz3Km7sQZFTdB996iG) for getting to know common flutter widgets.

Also, i'm going to use emotes to display importance of a review
laugh: not very important.
eyes: fairly important
-1 (thumbs down) : very important!

divy-arpit committed:
Add UI to display the list of organizations

This commit adds to the app, the much needed UI which would show  the user a list of all organizations of which he is a part.

naveennvrgup reviewed (changes requested):
1. Great job with the first PR @divy-arpit :fire: 
1. Please rename all `list of orgs` into `orgsList` :innocent: (folders, files, classes and variables)
1. Please extract the scaffolding into a common widget.
1. can reasonably size the text in the list tile
1. @werainkhatri  I have resolved the conversations which are resolved.

werainkhatri commented:
> 1. Great job with the first PR @divy-arpit :fire: 
> 1. Please rename all `list of orgs` into `orgsList` :innocent: (folders, files, classes and variables)
> 1. Please extract the scaffolding into a common widget.
> 1. can reasonably size the text in the list tile
> 1. @werainkhatri  I have resolved the conversations which are resolved.

https://github.com/ECellNitrr/teesco-app/pull/30#discussion_r432359064

actually, the naming convention for files and folders is supposed to be like this...😅

divy-arpit committed:
Improve naming of files according to conventions and font size

This commit improves the naming of files and the font size of text on the title tile.

werainkhatri commented on lib/screens/organization_list/organization_list_screen.dart:
```diff
@@ -0,0 +1,40 @@
+        body: Column(
+          children: <Widget>[
+            OrganizationCard(),
+            OrganizationCard(),
```
Keep the spelling of `Organisation` same everywhere

werainkhatri commented on lib/screens/organization_list/organization_list_screen.dart:
```diff
@@ -0,0 +1,40 @@
+import 'package:flutter/material.dart';
+import './widgets/organization_card.dart';
+/* this is the screen which will be displayed when the user wants to see the list of organisations of which he is a part
```
Try to use [doc comments](https://dart.dev/guides/language/effective-dart/documentation)
```suggestion
/// this is the screen which will be displayed when the user wants to see the list of organisations of which he is a part
```

werainkhatri commented on lib/screens/organization_list/widgets/organization_card.dart:
```diff
@@ -0,0 +1,48 @@
+import 'package:flutter/material.dart';
+
+class OrganizationCard extends StatelessWidget {
```
You can make this class more robust
create 2 final fields, String name and position
and accept them using a constructor.
Then you pass these parameters in the screen class

divy-arpit committed:
Minute changes in organization_card class

This commit adds a constructor to the class

naveennvrgup committed:
Merge branch 'dev' into list-of-orgs

naveennvrgup reviewed (approved):
1. Tested locally works as expected
1. All issues pointed out by @werainkhatri is been resolved
1. Great job  @divy-arpit :clap:
